pennylane/templates/subroutines/adder.py

```diff
@@ -22,54 +22,83 @@
+    `arXiv:2311.08555 <https://arxiv.org/abs/2311.08555>`_.
 
     .. note::
 
-        Note that :math:`x` must be smaller than :math:`mod` to get the correct result.
+        :math:`x` must be smaller than :math:`mod` to get the correct result.
 
+    .. seealso:: :class:`~.PhaseAdder` and :class:`~.OutAdder`.
 
-     Args:
-         k (int): the number that needs to be added
-         x_wires (Sequence[int]): the wires the operation acts on
-         mod (int): the modulus for performing the addition, default value is :math:`2^{len(x\_wires)}`
-         work_wires (Sequence[int]): the auxiliary wires to be used for performing the addition
+    Args:
+        k (int): the number that needs to be added
+        x_wires (Sequence[int]): the wires the operation acts on
+        mod (int): the modulus for performing the addition, default value is :math:`2^{\text{len(x_wires)}}`
+        work_wires (Sequence[int]): the two auxiliary wires to be used for performing the addition when :math:`mod \neq 2^{\text{len(x_wires)}}`
 
      **Example**
 
     This example computes the sum of two integers :math:`x=8` and :math:`k=5` modulo :math:`mod=15`.
 
-     .. code-block::
+    .. code-block::
 
-         x = 8
-         k = 5
-         mod = 15
+        x = 8
+        k = 5
+        mod = 15
 
-         x_wires =[0,1,2,3]
-         work_wires=[4,5]
+        x_wires =[0,1,2,3]
+        work_wires=[4,5]
 
-         dev = qml.device("default.qubit", shots=1)
-         @qml.qnode(dev)
-         def circuit(x, k, mod, x_wires, work_wires):
-             qml.BasisEmbedding(x, wires=x_wires)
-             qml.Adder(k, x_wires, mod, work_wires)
-             return qml.sample(wires=x_wires)
+        dev = qml.device("default.qubit", shots=1)
+        @qml.qnode(dev)
+        def circuit():
+            qml.BasisEmbedding(x, wires=x_wires)
+            qml.Adder(k, x_wires, mod, work_wires)
+            return qml.sample(wires=x_wires)
 
-     .. code-block:: pycon
+    .. code-block:: pycon
 
-         >>> print(circuit(x, k, mod,x_wires, work_wires))
-         [1 1 0 1]
+        >>> print(circuit())
+        [1 1 0 1]
 
     The result, :math:`[1 1 0 1]`, is the ket representation of
     :math:`8 + 5  \, \text{modulo} \, 15 = 13`.
+
+    .. details::
+        :title: Usage Details
+
+        This template takes as input two different sets of wires.
+
+        The first one is ``x_wires`` which is used
+        to encode the integer :math:`x < mod` in the computational basis. Therefore, we need at least
+        :math:`\lceil \log_2(x)\rceil` ``x_wires`` to represent :math:`x`. After performing the modular addition operation, the resulting integer
+        encoded in the computational basis can be as large as :math:`mod-1`. Hence, we need at least
+        :math:`\lceil \log_2(mod)\rceil` ``x_wires``
+        to represent all the possible results. Since :math:`x < mod` by definition, we just need at least :math:`\lceil \log_2(mod)\rceil` ``x_wires``.
```

Comment:
```suggestion
        to represent all the possible results. Since :math:`x < mod` by definition, we need at least :math:`\lceil \log_2(mod)\rceil` ``x_wires``.
```

Reply:
I've rewritten this paragraph to be more concise and clear